app/controllers/memes_controller.rb

```diff
@@ -65,6 +65,76 @@ def destroy
     end
   end
 
+  def upvote
+    @vote = Vote.where(meme_id: params[:id], user: current_user)
+
+    if @vote.count != 0
+      @vote_value = @vote[0].value
+      @vote.destroy_all
+
+      @created_upvote = false;
+      @deleted_downvote = false;
+
+      if @vote_value == false
+        Vote.create(meme_id: params[:id], user: current_user, value: true)
+        @created_upvote = true;
+        @deleted_downvote = true;
+      end
+    else
+      Vote.create(meme_id: params[:id], user: current_user, value: true)
+      @created_upvote = true;
+      @deleted_downvote = false;
+    end
+
+    # Return value
```

Comment:
Instead of returning two booleans like this, just return -1, 0, or 1. It'll simply your logic a lot in the JS. 
* If -1, unset any color from upvote button (there might not be any) and set color on downvote button.
* If 0, unset color from both buttons
* If 1, unset color from downvote button, set color on upvote button